core/tests/test_utils.py

```diff
@@ -1076,6 +1288,45 @@ def swap(self, obj, attr, newvalue):
         finally:
             setattr(obj, attr, original)
 
+    @contextlib.contextmanager
+    def swap_conditionally(
+            self, obj, attr, new_function=lambda *_, **__: None,
+            condition=None):
+        """Swap the obj.attr function to return a value when a condition is met.
+
+        When the condition is not met, then the original function will be called
+        instead.
+
+        This function can be used when the target being swapped is used by
+        unrelated code. For example, os.path.exists() is used extensively by
+        psutil to manage processes. If we swap it to _always_ return True or
+        False, then the process management will result in errors.
+
+        Args:
+            obj: *. The object whose attribute will be swapped.
+            attr: str. The attribute of the object to swap.
+            new_function: callable. The function called when the condition
+                returns True. By default, the replacement function accepts any
+                arguments and returns None.
+            condition: callable|None. A predicate function which returns True or
```

Comment:
Why would we use swap_conditionally() with a None condition? IMO that shouldn't be allowed and this function should only be used when necessary.

Reply:
Obsolete.